### PR TITLE
Housekeeping cleanups and improvements before adding CI

### PR DESCRIPTION
**Pull request by lyarwood**
/cc @ksimon1 

**What this PR does / why we need it**:

This series starts to cleanup the repository ahead of introducing meaningful CI.

The bulk of the changes are around using `kubeconform` to validate the generated definitions against the KubeVirt API.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,23 +1,27 @@
 SHELL=/bin/bash
 
-all: lint generate
+# Run `make schema` when updating this version and commit the created files.
+# TODO(lyarwood) - Host the expanded JSON schema elsewhere under the kubevirt namespace
+export KUBEVIRT_VERSION = main
```
Comment (ksimon1) [inline Makefile:5]: is this variable used somewhere?

File: README.md [hunk 2]
```diff
@@ -1,172 +1,244 @@
...
+windows.2k12.virtio
+---
+VirtualMachineClusterPreference
+windows.2k16
+---
+VirtualMachineClusterPreference
+windows.2k16.virtio
+---
+VirtualMachineClusterPreference
+windows.2k19
```
Comment (ksimon1) [inline README.md:131]: Its missing win2k22

File: README.md [hunk 3]
```diff
@@ -1,172 +1,244 @@
...
+centos.7
+---
+VirtualMachinePreference
+centos.7.desktop
+---
+VirtualMachinePreference
+centos.8
+---
+VirtualMachinePreference
+centos.8.desktop
```
Comment (ksimon1) [inline README.md:170]: what about centos stream?

File: Makefile [hunk 4]
```diff
@@ -1,23 +1,27 @@
...
-	kustomize build VirtualMachineInstancetypes >> common-instancetypes-bundle.yaml
-	echo "---" > common-clusterinstancetypes-bundle.yaml
-	kustomize build VirtualMachineClusterInstancetypes >> common-clusterinstancetypes-bundle.yaml
-	echo "---" > common-preferences-bundle.yaml
-	kustomize build VirtualMachinePreferences >> common-preferences-bundle.yaml
-	echo "---" > common-clusterpreferences-bundle.yaml
-	kustomize build VirtualMachineClusterPreferences >> common-clusterpreferences-bundle.yaml
+	./scripts/lint.sh
+
+generate:
```
Comment (ksimon1) [inline Makefile:12]: It would be nice if makefile downloads all required bins for user

File: scripts/generate.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Comment (ksimon1) [inline scripts/generate.sh:1]: Do you plan to have the file names with version?

File: schemas/main/accesscredential.json [hunk 6]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (ksimon1) [inline schemas/main/accesscredential.json:1]: Is this folder going to be regenerated with each PR?

File: Makefile (hunk 1 above)
Comment (lyarwood) [reply Makefile:5]: Yes both the schema and validate scripts use it.

```
$ git grep -l KUBEVIRT_VERSION scripts/
scripts/schema.sh
scripts/validate.sh
```

File: README.md (hunk 2 above)
Comment (lyarwood) [reply README.md:131]: ACK i'll add this in a separate PR.

File: README.md (hunk 3 above)
Comment (lyarwood) [reply README.md:170]: ACK i'll add this in a separate PR.

File: scripts/generate.sh (hunk 5 above)
Comment (lyarwood) [reply scripts/generate.sh:1]: Yeah I've not started drafting the release logic yet but I'd assume we would version the bundles yes, I'm not sure that we'd version the output of this script however.

File: schemas/main/accesscredential.json (hunk 6 above)
Comment (lyarwood) [reply schemas/main/accesscredential.json:1]: It isn't at present, I've flipped flopped on if this should be against the main branch, latest tag or a random commit tbh.

The benefit of following main is that we can immediately consume new things added in main, the downside is that we'd need to somehow limit this to a specific tag/commit when we generate a release from common-instancetypes to ensure CI continues to work.

File: README.md (hunk 2 above)
Comment (lyarwood) [reply README.md:131]: https://github.com/kubevirt/common-instancetypes/pull/4

File: README.md (hunk 3 above)
Comment (lyarwood) [reply README.md:170]: https://github.com/kubevirt/common-instancetypes/pull/3

**Comment by lyarwood** [comment]
The big open question here is which release we target for the KubeVirt schema. The options I believe we have are: 

- Track `main` from common-instancetypes `main` and then a specific commit/tag per release of `common-instancetypes`
- Track the latest KubeVirt release (`v0.58.0` currently ) in our `main` so releases of `common-instancetypes` are tied to a specific KubeVirt release.
- Track `main` on both our `main` and releases of `common-instancetypes`, this is going to get out of sync quickly.

There's also the open question around if we should even host these schema files in this project or have another that automatically updates and serves the files as I assume more people would like to use `kubeconform` to validate their definitions.

**Comment by lyarwood** [comment]
/approve
/lgtm

This series now looks ready ahead of prow jobs being introduced. I'm the only approver at the moment so while I hate approving my own thing it's all I can do, I'll follow up with others about joining me so we avoid this in the future.

**Comment by lyarwood** [comment]
Ah I can't approve my own change anyway and we need the `OWNERS` file first before I'm even recognised as an approver. I'll merge this myself then.

**Comment by ksimon1** [comment]
it is not about OWNER file. Kubevirt bot will complain about lgtming your own PR even if you approver.

**Pull request merged**